src/workerd/api/pyodide/pyodide.h

```diff
@@ -178,10 +186,11 @@ class ArtifactBundler : public jsg::Object {
     auto& cb = KJ_REQUIRE_NONNULL(uploadMemorySnapshotCb);
     hasUploaded = true;
     auto& context = IoContext::current();
-    return context.awaitIo(js, cb(kj::mv(snapshot)));
+    return context.awaitIo(js, cb({
+        .snapshot = kj::mv(snapshot), .importedModulesList = {}}));
   };
 
-  void storeMemorySnapshot(jsg::Lock& js, kj::Array<kj::byte> snapshot) {
+  void storeMemorySnapshot(jsg::Lock& js, MemorySnapshotResult snapshot) {
```

Comment:
Instead of a `MemorySnapshotResult` I'd recommend that this take two separate arguments, the snapshot and the list of strings. That way it will be backwards compatible to the previous api and cause less trouble with #2430.

Reply:
I'll see how painful rebasing is.